tests/test_percolator.py

```diff
@@ -11,257 +12,77 @@
-from invenio_oaiserver.errors import OAISetSpecUpdateError
 from invenio_oaiserver.models import OAISet
-from invenio_oaiserver.receivers import after_delete_oai_set, \
-    after_insert_oai_set, after_update_oai_set
+from invenio_oaiserver.query import OAINoRecordsMatchError, get_records
+from invenio_oaiserver.receivers import after_update_oai_set
 
 
-def test_search_pattern_change(app, without_oaiset_signals, schema):
-    """Test search pattern change."""
-    record0 = create_record(app, {
-        '_oai': {'sets': ['a']}, 'title_statement': {'title': 'Test0'},
+@pytest.fixture()
+def test0(app, without_oaiset_signals, schema):
+    _ = create_record(app, {
+        'title_statement': {'title': 'Test0'},
         '$schema': schema
     })
-    rec_uuid = record0.id
-    oaiset = OAISet(spec="a", search_pattern="title_statement.title:Test0")
+    current_search.flush_and_refresh("records")
+
+
+def create_oaiset(name, title_pattern):
+    oaiset = OAISet(
+        spec=name,
+        search_pattern=f"title_statement.title:{title_pattern}"
+    )
     db.session.add(oaiset)
     db.session.commit()
     run_after_insert_oai_set()
-    current_search.flush_and_refresh('records')
-    record = Record.get_record(rec_uuid)
-    assert record['_oai']['sets'] == ['a']
-
-    # change search pattern: record0 will not inside it anymore
-    oaiset = OAISet.query.first()
-    oaiset.search_pattern = 'title_statement.title:Test1'
-    db.session.merge(oaiset)
-    db.session.commit()
-    after_update_oai_set(None, None, oaiset)
-    current_search.flush_and_refresh('records')
-    record = Record.get_record(rec_uuid)
-    record.commit()
-    assert record['_oai']['sets'] == []
 
+    return oaiset
 
-def test_populate_oaisets(app, without_oaiset_signals, schema):
-    """Populate OAISets."""
-    def create_oaiset(**kwargs):
-        oaiset = OAISet(**kwargs)
-        db.session.add(oaiset)
-        db.session.commit()
-        return oaiset
 
-    a = create_oaiset(spec='a')
-    create_oaiset(spec='b')
-    create_oaiset(
-        spec="e", search_pattern="title_statement.title:Test2 OR "
-        "title_statement.title:Test3")
-    create_oaiset(spec="c", search_pattern="title_statement.title:Test0")
-    create_oaiset(spec="d", search_pattern="title_statement.title:Test1")
-    f = create_oaiset(spec="f", search_pattern="title_statement.title:Test2")
-    create_oaiset(spec="g")
-    create_oaiset(spec="h")
-    i = create_oaiset(spec="i", search_pattern="title_statement.title:Test3")
-    j = create_oaiset(spec="j with space",
-                      search_pattern="title_statement.title:Test4")
-    # Note below: brackets around AND search query are required
-    create_oaiset(spec="math",
-                  search_pattern="(title_statement.title:foo AND genre:math)")
-    create_oaiset(spec="nonmath",
-                  search_pattern="(title_statement.title:foo AND -genre:math)")
+def test_set_with_no_records(without_oaiset_signals, schema):
```

Comment:
The previous tests were hard to read and I think had multiple checks for the same thing. Plus they make use of `add_record` and `remove_record` for manually managed sets, which we cannot do now with the dynamic ones.
I tried to bring back the tests that make sense and split them by use case.